src/SparkTestBase.sol

```diff
@@ -496,66 +680,44 @@ abstract contract SparkEthereumTestBase is SparklendTestBase {
+   function _assertRateLimit(
+       bytes32 key,
         uint256 maxAmount,
         uint256 slope
     ) internal {
-        IRateLimits.RateLimitData memory rateLimit = IRateLimits(Ethereum.ALM_RATE_LIMITS).getRateLimitData(key);
+        ChainId currentChain = ChainIdUtils.fromUint(block.chainid);
+        IRateLimits rateLimitsContract;
+        if(currentChain == ChainIdUtils.Ethereum()) rateLimitsContract = IRateLimits(Ethereum.ALM_RATE_LIMITS);
+        else if(currentChain == ChainIdUtils.Base()) rateLimitsContract = IRateLimits(Base.ALM_RATE_LIMITS);
+        else require(false, "ALM/executing on unknown chain");
+
+        IRateLimits.RateLimitData memory rateLimit = rateLimitsContract.getRateLimitData(key);
         assertEq(rateLimit.maxAmount,   maxAmount);
         assertEq(rateLimit.slope,       slope);
         assertEq(rateLimit.lastAmount,  maxAmount);
         assertEq(rateLimit.lastUpdated, block.timestamp);
     }
 }
 
-abstract contract SparkGnosisTestBase is SparklendTestBase {
-
-    constructor() {
-        executor = Gnosis.AMB_EXECUTOR;
-        domain   = 'Gnosis';
-
-        poolAddressesProviderRegistry = IPoolAddressesProviderRegistry(Gnosis.POOL_ADDRESSES_PROVIDER_REGISTRY);
-    }
-
-    function executePayload(address payloadAddress) internal override {
-        require(Address.isContract(payloadAddress), "PAYLOAD IS NOT A CONTRACT");
-        vm.prank(executor);
-        IExecutor(executor).executeDelegateCall(
-            payloadAddress,
-            abi.encodeWithSignature('execute()')
-        );
-    }
-
-}
-
-abstract contract SparkBaseTestBase is SparkTestBase {
-
-    constructor() {
-        executor = Base.SPARK_EXECUTOR;
-        domain   = 'Base';
-    }
-
-    function executePayload(address payloadAddress) internal override {
-        require(Address.isContract(payloadAddress), "PAYLOAD IS NOT A CONTRACT");
-        vm.prank(executor);
-        IExecutor(executor).executeDelegateCall(
-            payloadAddress,
-            abi.encodeWithSignature('execute()')
-        );
-    }
-
-    function _assertRateLimit(
-        bytes32 key,
-        uint256 maxAmount,
-        uint256 slope
-    ) internal {
-        IRateLimits.RateLimitData memory rateLimit = IRateLimits(Base.ALM_RATE_LIMITS).getRateLimitData(key);
-        assertEq(rateLimit.maxAmount,   maxAmount);
-        assertEq(rateLimit.slope,       slope);
-        assertEq(rateLimit.lastAmount,  maxAmount);
-        assertEq(rateLimit.lastUpdated, block.timestamp);
+/// @dev convenience contract meant to be the single point of entry for all
+/// spell-specifictest contracts
+abstract contract SparkTestBase is AdvancedLiquidityManagementTests, SparkEthereumTests, CommonSpellAssertions {
+    using DomainHelpers for StdChains.Chain;
+    using DomainHelpers for Domain;
+
+    // cant really instruct the compiler to simply use the SparklendTests
+    // implementation, so I copied it.
```

Comment:
I tried to minimize load on the test writers with this, alternatives are:
- making the `AdvancedLiquidityManagementTests` extend `SparklendTests`: would allow me to remove the modifier, but wouldn't feel correct since ALM does not depend on sparklend
- only _declare_ it on `SpellRunner` as virtual, then let this implementation be the only one in the codebase: would avoid duplication, but would also force us to make all contracts abstract (which they already are) when there's no real reason for it (but also wouldn't cost us anything)
- run the `loadPoolContext` call somewhere else: couldn't think of a way that is less fragile than this. Perhaps a virtual  `setupDomains`? but in that case setup would be done for a single chain (presumabley mainnet), and a test using the `onChain` modifier to run on Gnosis would require a `loadPoolContext` call in the test